### PR TITLE
Error message should match x, y, z arguments

### PR DESCRIPTION
**Pull request by abarciauskas-bgse**
It's a bit confusing when the error message has a different order than the arguments, so this is a small change to the error message.

### CODE REVIEW

File: rio_tiler/io/xarray.py [hunk 1]
```diff
@@ -232,7 +232,7 @@ def tile(
             reproject_method = resampling_method
 
         if not self.tile_exists(tile_x, tile_y, tile_z):
-            raise TileOutsideBounds(f"Tile {tile_z}/{tile_x}/{tile_y} is outside bounds")
+            raise TileOutsideBounds(f"Tile {tile_x}/{tile_y}/{tile_z} is outside bounds")
```
Comment (vincentsarago) [inline rio_tiler/io/xarray.py:235]: maybe we should print something like `Tile(x={tile_x}, y={tile_y}, z={tile_z})`

Comment (vincentsarago) [reply rio_tiler/io/xarray.py]: would you also be able to update https://github.com/cogeotiff/rio-tiler/blob/main/rio_tiler/io/rasterio.py#L344-L347, https://github.com/cogeotiff/rio-tiler/blob/91095a27e663c76f2dcfdac64d4364ad9f9b9585/rio_tiler/io/base.py#L1037-L1040 and https://github.com/cogeotiff/rio-tiler/blob/91095a27e663c76f2dcfdac64d4364ad9f9b9585/rio_tiler/io/base.py#L481-L484 🙏

**Comment by vincentsarago** [comment]
🙏 thanks @abarciauskas-bgse

**Comment by abarciauskas-bgse** [comment]
Thanks @vincentsarago for making the additional changes and merging. I was going to write a test for the error message but 👌🏽 with me to forego that for now.

**Pull request merged**